iron-jsonp-library.d.ts

```diff
@@ -0,0 +1,90 @@
+/**
+ * DO NOT EDIT
+ *
+ * This file was automatically generated by
+ *   https://github.com/Polymer/gen-typescript-declarations
+ *
+ * To modify these typings, edit the source file(s):
+ *   iron-jsonp-library.html
+ */
+
+/// <reference path="../polymer/types/polymer.d.ts" />
+
+declare namespace Polymer {
+
+  /**
+   * `Polymer.IronJsonpLibraryBehavior` loads a jsonp library.
+   * Multiple components can request same library, only one copy will load.
+   *
+   * Some libraries require a specific global function be defined.
+   * If this is the case, specify the `callbackName` property.
+   *
+   * You should use an HTML Import to load library dependencies
+   * when possible instead of using this element.
+   */
+  interface IronJsonpLibraryBehavior {
+
+    /**
+     * True if library has been successfully loaded
+     */
+    readonly libraryLoaded: boolean|null|undefined;
+
+    /**
+     * Not null if library has failed to load
+     */
+    readonly libraryErrorMessage: string|null|undefined;
+    _libraryUrlChanged(libraryUrl: any): void;
+    _libraryLoadCallback(err: any, result: any): void;
+
+    /**
+     * loads the library, and fires this.notifyEvent upon completion
+     */
+    _loadLibrary(): void;
+    ready(): void;
+  }
+
+  const IronJsonpLibraryBehavior: object;
```

Comment:
Polymer.IronJsonpLibraryBehvaior is an interface. Is this const needed as well?

Reply:
Yep, kinda funky. The interface is an abstract thing representing how the behavior will augment an element, while the const represents the actual behavior object that you pass to e.g. the mixinBehaviors function. Some more context here: https://github.com/Polymer/gen-typescript-declarations/blob/master/src/gen-ts.ts#L368